File: seed/ruby-model/trace/lib/seed_trace_client/submission/types/execution_session_status.rb
# frozen_string_literal: true

module SeedTraceClient
  class Submission
    class ExecutionSessionStatus
      CREATING_CONTAINER = "CREATING_CONTAINER"
      PROVISIONING_CONTAINER = "PROVISIONING_CONTAINER"
      PENDING_CONTAINER = "PENDING_CONTAINER"
      RUNNING_CONTAINER = "RUNNING_CONTAINER"
      LIVE_CONTAINER = "LIVE_CONTAINER"
      FAILED_TO_LAUNCH = "FAILED_TO_LAUNCH"
    end
  end
end

File: seed/rust-model/trace/src/execution_session_status.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub enum ExecutionSessionStatus {
    #[serde(rename = "CREATING_CONTAINER")]
    CreatingContainer,
    #[serde(rename = "PROVISIONING_CONTAINER")]
    ProvisioningContainer,
    #[serde(rename = "PENDING_CONTAINER")]
    PendingContainer,
    #[serde(rename = "RUNNING_CONTAINER")]
    RunningContainer,
    #[serde(rename = "LIVE_CONTAINER")]
    LiveContainer,
    #[serde(rename = "FAILED_TO_LAUNCH")]
    FailedToLaunch,
}